ext/ImplicitDifferentiationForwardDiffExt.jl

```diff
@@ -39,11 +41,14 @@ function (implicit::ImplicitFunction)(
     pfB = pushforward_function(backend, _x -> conditions(_x, y, z; kwargs...), x)
     A_op = LinearOperator(R, m, m, false, false, PushforwardMul!(pfA, size(y)))
     B_op = LinearOperator(R, m, n, false, false, PushforwardMul!(pfB, size(x)))
+    A = presolver(A_op, x, y)
 
     dy = map(1:N) do k
         dₖx_vec = vec(partials.(x_and_dx, k))
-        dₖy_vec, stats = linear_solver(A_op, B_op * dₖx_vec)
-        dₖy_vec .*= -1
+        Bdx = vec(similar(y))
+        mul!(Bdx, B_op, dₖx_vec)
+        dₖy_vec, stats = linear_solver(A, Bdx)
+        dₖy_vec = -dₖy_vec
```

Comment:
Should dispatch to do this inplace when it is a `Vector` and out-of-place when `SVector`.